File: tasks_26/solutions/26-44.py
with open("26-44_.txt") as F:
   N = int(F.readline())
   data = []
   for i in range(N):
     data.append( int(F.readline()) )

data.sort()
last = 500
ma = 0
discount = 0
while data:
  chunk = [x for x in data if x <= last]
  if chunk:
    mid = len(chunk)//2
    if mid > 0:
      discount += sum(chunk[:mid])*0.5
      costMax = chunk[mid-1]*0.5
    data = data[len(chunk):]
  last+= 500

print( int(discount), int(costMax) )

File: tasks_26/solutions/26-44.pas
##
assign(input, '26-44_.txt');
var N := ReadInteger;
var data := ReadArrInteger(N);

data.Sort;

var last := 500;
var discount := 0.0;
var costMax := 0.0;
while data.Count > 0 do begin
  var chunk := data.Where(x->x<=last).ToArray;
  if chunk.Count > 0 then begin
    var mid := chunk.Count div 2;
    if mid > 0 then begin
      discount += 0.5*chunk[:mid].Sum;
      costMax := 0.5*chunk[mid-1];
    end;  
    data := data.Where(x->x>last).ToArray;   
  end;
  last += 500;
end; 

Println( trunc(discount), trunc(costMax) );